core/src/main/java/org/apache/metamodel/util/WildcardPattern.java

```diff
@@ -32,13 +32,15 @@
 public final class WildcardPattern implements Serializable {
 
 	private static final long serialVersionUID = 857462137797209624L;
+	private final boolean _startsWithDelim;
+	private final boolean _endsWithDelim;
 	private String _pattern;
 	private char _wildcard;
-	private boolean _endsWithDelim;
 
 	public WildcardPattern(String pattern, char wildcard) {
 		_pattern = pattern;
 		_wildcard = wildcard;
+		_startsWithDelim = (_pattern.charAt(0) == _wildcard);
```

Comment:
This will throw IndexOutOfBoundsException if an empty string is passed as the pattern.